roofit/roofitcore/src/RooCurve.cxx

```diff
@@ -118,13 +118,21 @@ RooCurve::RooCurve(const RooAbsReal &f, RooAbsRealLValue &x, Double_t xlo, Doubl
 
   // calculate the points to add to our curve
   Double_t prevYMax = getYAxisMax() ;
-  list<Double_t>* hint = f.plotSamplingHint(x,xlo,xhi) ;
-  addPoints(*funcPtr,xlo,xhi,xbins+1,prec,resolution,wmode,nEvalError,doEEVal,eeVal,hint);
-  if (_showProgress) {
-    ccoutP(Plotting) << endl ;
-  }
-  if (hint) {
-    delete hint ;
+  if(xbins > 0){
+    list<Double_t>* hint = f.plotSamplingHint(x,xlo,xhi) ;
+    addPoints(*funcPtr,xlo,xhi,xbins+1,prec,resolution,wmode,nEvalError,doEEVal,eeVal,hint);
+    if (_showProgress) {
+      ccoutP(Plotting) << endl ;
+    }
+    if (hint) {
+      delete hint ;
+    }
+  } else {
+    xbins = x.numBins();
```

Comment:
Is there a need to reset `xbins` to the new value here? Wouldn't a local variable that can be optimised out be sufficient?
Something like:
```
const size_t nBinsX = x.numBins();
<for loop until nBinsX>
```

Reply:
good suggestion, agreed.